flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adapter/DefaultExecutionVertex.java

```diff
@@ -18,35 +18,52 @@
 
 package org.apache.flink.runtime.scheduler.adapter;
 
+import org.apache.flink.annotation.VisibleForTesting;
 import org.apache.flink.runtime.execution.ExecutionState;
+import org.apache.flink.runtime.jobgraph.IntermediateResultPartitionID;
+import org.apache.flink.runtime.scheduler.strategy.ConsumedPartitionGroup;
 import org.apache.flink.runtime.scheduler.strategy.ExecutionVertexID;
 import org.apache.flink.runtime.scheduler.strategy.SchedulingExecutionVertex;
 
-import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 import java.util.function.Supplier;
 
+import static org.apache.flink.runtime.scheduler.SchedulerUtils.createFlattenIterator;
 import static org.apache.flink.util.Preconditions.checkNotNull;
 
 /** Default implementation of {@link SchedulingExecutionVertex}. */
 class DefaultExecutionVertex implements SchedulingExecutionVertex {
 
     private final ExecutionVertexID executionVertexId;
 
-    private final List<DefaultResultPartition> consumedResults;
-
     private final List<DefaultResultPartition> producedResults;
 
     private final Supplier<ExecutionState> stateSupplier;
 
+    private final List<ConsumedPartitionGroup> consumedPartitionGroups;
+
+    private final Map<IntermediateResultPartitionID, DefaultResultPartition> resultPartitionsById;
```

Comment:
I'm wondering that since we initialize `Map<IntermediateResultPartitionID, DefaultResultPartition>` in `DefaultExecutionTopology#computeExecutionGraphIndex`, would the map be garbage collected if we only pass the `Function` as the parameter?

Reply:
I think this lamda will have a reference to the map so it will not be GCed.
ref: https://tanbt.medium.com/java-lambdas-and-the-garbage-collector-part-1-da934e11518a

Reply:
Thanks for proving this reference. I've resolved it.